src/util/ImportStringBuilder.ts

```diff
@@ -0,0 +1,230 @@
+import {ClassDecl} from "../ast/ClassDecl";
+import {PathTable} from "./PathTable";
+import {VarList} from "../ast/VarList";
+import {AstNode} from "../ast/AstNode";
+import {InterfaceDecl} from "../ast/InterfaceDecl";
+import FuncDecl from "../ast/FuncDecl";
+
+export class ImportStringBuilder {
+
+    /**
+     * Extracts the non primitive types used by a class/interface declaration then
+     * returns a series of import strings to be located at the top of a class/interface declaration.
+     *
+     * e.g. import {VarList} from "../ast/VarList";
+     *      import {PathTable} from "./PathTable";
+     *      import {ClassDecl} from "../ast/ClassDecl";
+     *
+     * @param node              the class/interface declaration doing the importing
+     */
+    public static getImportsString(node: AstNode): string {
+        if (node instanceof ClassDecl) {
+            return this.getClassImportsString(node);
+        } else if (node instanceof InterfaceDecl) {
+            return this.getInterfaceImportsString(node);
+        } else {
+            const msg: string = `Given node: ${node} cannot have imports string`;
+            console.warn(msg);
+            throw new Error(msg);
+        }
+    }
+
+    /**
```

Comment:
For the places where we have repeated logic (e.g. getClassImportsString/ getInterfaceImportsString, is it possible to just consolidate these methods to one using a union type? e.g.:
`private static getImportsString(decl: ClassDecl | InterfaceDecl): string `

Reply:
I didnt know this was a thing, ill look into it!

Reply:
Ok so they have to be separate because the way we get fields from one and the other is different. Class have lists of FieldDecl while Interface only has one FieldDecl. The logic is different so down the line they would also have to be separated.

Reply:
I see, oki doki- thanks for looking into it!

Reply:
Actually I looked at it more and there were a couple of things I could do. Check it out - I tried to reduce repeated logic as much as possible.